packages/client/hmi-client/src/components/model/petrinet/tera-line-graphic.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+	<div :id="graphicId">
+		<svg :height="parentHeight">
+			<line x1="0" :y1="parentHeight / 2" :x2="width" :y2="parentHeight / 2" stroke="black" />
+			<rect
+				v-if="distribution.type === DistributionType.Uniform"
+				:x="rectX"
+				:y="parentHeight / 2 - 2"
+				:width="rectWidth"
+				rx="2"
+				height="4"
+				stroke="gray"
+				fill="gray"
+			/>
+			<circle v-else :cx="circlePoint" :cy="parentHeight / 2" r="4" stroke="black" fill="black" />
+		</svg>
+	</div>
+</template>
+
+<script setup lang="ts">
+import * as d3 from 'd3';
+import { onMounted, ref } from 'vue';
+import { DistributionType } from '@/services/distribution';
+
+const props = defineProps<{
+	id: any;
+	maxValue: any;
+	minValue: any;
+	distribution: any;
+}>();
+const graphicId = 'graphic-'.concat(props.id);
+const parentWidth = ref<number>(0);
+const parentHeight = ref<number>(0);
+const width = ref<string>('100%');
+const rectWidth = ref<number>(0);
+const rectX = ref<number>(0);
+const circlePoint = ref<number>(0);
+const range = ref<Array<number>>([1, 100]);
+
+const max: number = props.maxValue;
+const min: number = props.minValue;
+
+const isUniformType = () => props.distribution.type === DistributionType.Uniform;
+
+onMounted(() => {
+	parentWidth.value = document.getElementById(graphicId)?.getBoundingClientRect().width ?? 0;
+	parentHeight.value = document.getElementById(graphicId)?.getBoundingClientRect().height ?? 0;
```

Comment:
Generally speaking, don't use `id` attribute in components that are meant to be reused.

Moreover don't use document-level query selectors, cause you never know what you are gonna get. 

If you need to get the width/height you can do it through a element-ref in vue, it would also be a bit more idiomatic with the other things in the code base.